### ISSUE TITLE
i18n support for scaffolds?

### ISSUE THREAD

**Issue by yshmarov** [rails/rails#51405]
I think it would be great if there was an option to generating scaffolds with i18n support by default.

So when I run `rails g scaffold post title body:text` it would also inject i18n defaults into the `en.yml` file and into controllers & views.

Something like

```yml
# config/locales/en.yml

en:
  posts:
    create:
      notice: Post was successfully created.
    update:
      notice: Post was successfully updated.
    destroy:
      notice: Post was successfully destroyed.
  activerecord:
    attributes:
      posts:
        title: Title
        body: Body
```

```diff
# app/controllers/posts_controller.rb
-        format.html { redirect_to post_url(@post), notice: "Post was successfully created." }
+         format.html { redirect_to post_url(@post), notice: t('.notice') }
```

```diff
# app/views/posts/_post.html.erb
<div id="<%= dom_id post %>">
  <p class="my-5">
-    <strong class="block font-medium mb-1">Body:</strong>
+    <strong class="block font-medium mb-1"><%= t('activerecord.attributes.posts.body) %>:</strong>
```

What do you think?

**Comment by skipkayhil** [comment]
Thank you for the suggestion but we reserve the issue tracker for bugs only. For feature requests/discussion the best place would be the [forums](https://discuss.rubyonrails.org/)

**Comment by JoeDupuis** [comment]
@yshmarov, [I've suggested this before](https://discuss.rubyonrails.org/t/i18n-in-scaffolded-controller-and-views/75142), but ended up not writing the PR, because someone on Rails core (I believe DHH), mentioned that the scaffold are intentionally "ugly" or bare. They are not intended to be used as is. You are expected to modify them heavily. 

In practice, I found that a lot of those modifications where *very* repetitive, so I wrote a set of generators that handle it for me. 

I think that there is a case to be made for upstreaming something like this. Although, it would likely be an an uphill battle. 

I rely a lot less on scaffolds than I use to, but I think there is good reasons to use the I18n layer even without translating the app. It let you separate copy from the code. 

Take this controller action for example:

```ruby
def update
    if @book.update(book_params)
      redirect_to @book, notice: success_message(@book)
    else
      render :edit, status: :unprocessable_entity
    end
end
```
and the helper implementation:

```ruby
  def success_message model = nil
    args = {}
    args[:name] = model.model_name.human if model
    t(".success", default: :"application.#{action_name}.success", **args)
  end
```

This lets me 
- have a generic success message that adapts automatically to the model name. 
- I can change the model name showed to the user independently from how it is called in the code. 
- I can write custom success message per model if needed.
- The controller doesn't have to worry about copy. It is only focused on returning a success message. It doesn't have to care what the success message is.

I'll try to post my generators soon™ and I'll open the discussion again.

**Comment by jpgeek** [comment]
@joeDupuis  @yshmarov
I have also pushed for this.  An additional issue is that there are no related translations in RailsI18n.  I have suggested adding them but the namespacing falls in a no-mans-land between I18n and the scaffold structures so has been refused. 
[rails scaffold change request](https://discuss.rubyonrails.org/t/proposal-i18n-translations-for-scaffold-templates/81777)
[Rails I18n issue](https://github.com/svenfuchs/rails-i18n/issues/1008)
